tools/@o3r/mcp/src/tools/find-repositories-using-otter.ts

```diff
@@ -0,0 +1,184 @@
+import {
+  existsSync,
+} from 'node:fs';
+import {
+  mkdir,
+  readFile,
+  writeFile,
+} from 'node:fs/promises';
+import {
+  resolve,
+} from 'node:path';
+import type {
+  McpServer,
+} from '@modelcontextprotocol/sdk/server/mcp.js';
+import {
+  Octokit,
+} from '@octokit/rest';
+import {
+  logger,
+} from '../utils/logger';
+
+async function listOrgRepos(octokit: Octokit) {
+  const repos = (await octokit.paginate(octokit.repos.listForOrg, {
+    org: process.env.GITHUB_ORG as string,
+    per_page: 100,
+    type: 'all',
+    sort: 'updated' // Prioritize recently updated repositories
+  })).filter((repo) => !repo.archived && !repo.fork);
+  logger.info(`Found ${repos.length} repositories in the organization ${process.env.GITHUB_ORG}`);
+  return repos;
+}
+
+type Repository = Awaited<ReturnType<typeof listOrgRepos>>[number];
+
+function findPackageJsonFiles(octokit: Octokit) {
+  return async (repository: Repository) => {
+    const owner = process.env.GITHUB_ORG as string;
+    const repo = repository.name;
+    const { data: { commit: { sha } } } = await octokit.repos.getBranch({
+      owner,
+      repo,
+      branch: repository.default_branch || 'main'
+    });
+    const { data: { tree } } = await octokit.git.getTree({
+      owner,
+      repo,
+      tree_sha: sha
+    });
+    return tree.filter((t) => t.type === 'blob' && t.path.endsWith('package.json'));
+  };
+}
+
+function dependsOnOtter(octokit: Octokit) {
+  return async (repository: Repository, packageJsonPath: string) => {
+    const owner = process.env.GITHUB_ORG as string;
+    const repo = repository.name;
+    const { data } = await octokit.repos.getContent({
+      owner,
+      repo,
+      path: packageJsonPath
+    });
+    if (Array.isArray(data) || data.type !== 'file') {
+      throw new Error('Unexpected content response structure');
+    }
+    const content = Buffer.from(data.content, 'base64').toString('utf8');
+    const packageJson = JSON.parse(content) as { dependencies?: Record<string, string>; devDependencies?: Record<string, string> };
+    return [
+      ...Object.keys(packageJson.devDependencies || {}),
+      ...Object.keys(packageJson.dependencies || {})
+    ].some((dep) => /^@(o3r|ama)/.test(dep));
```

Comment:
```suggestion
    ].some((dep) => /^@(o3r|ama-sdk|ama-mfe|ama-styling)\//.test(dep));
```
to avoid looking into projects such @o3rpouet or @amande

Reply:
@amazing

Reply:
or to avoid adding a new scope we got a new one:

```suggestion
    ].some((dep) => /^@(o3r|ama)[/-]/.test(dep));
```

Reply:
I change it to `/^@(o3r|ama-\w+)\//`